frozen-abi/Cargo.toml

```diff
@@ -13,22 +13,34 @@ edition = "2021"
 bs58 = "0.4.0"
 bv = { version = "0.11.1", features = ["serde"] }
 lazy_static = "1.4.0"
-log = "0.4.17"
+log = { version = "0.4.17", features = ["std"] }
 once_cell = "1.12.0"
-serde = "1.0.138"
+serde = { version = "1.0", features = ["derive", "rc"] }
 serde_bytes = "0.11"
-serde_derive = "1.0.103"
+serde_derive = "1.0"
+serde_json = "1.0"
 sha2 = "0.10.2"
 solana-frozen-abi-macro = { path = "macro", version = "=1.11.3" }
 thiserror = "1.0"
 
 [target.'cfg(not(target_os = "solana"))'.dependencies]
+ahash = { version = "0.7.6", features = ["default", "std"] }
+blake3 = { version = "1.3.1", features = ["digest", "traits-preview"] }
+block-buffer = { version = "0.9.0", features = ["block-padding"] }
+byteorder = { version = "1.4.3", features = ["default", "i128", "std"] }
+cc = { version = "1.0.67", features = ["jobserver", "parallel"] }
```

Comment:
When we build `programs/sbf/rust/*` crates  https://github.com/dmakarov/solana/blob/3bc670a47b5a39eb37983533aab345491eab3d75/ci/test-stable.sh#L103-L110 , we expect to be able to reuse previously built dependencies.  It happens that transitive dependencies are added with some variations to the dependency tree of every crate in that directory.  Adding ahash and blake3 to a common dependency of all these crates forces cargo to choose specific versions and feature sets of the transitive dependencies that satisfy dependency requirements of all `programs/sbf/rust/*` crates so that the number of rebuilds is minimized.  If I'm not mistaken ahash and blake3 are transitive dependencies of frozen-abi, and this change was an attempt to force cargo to use specific versions and feature sets of these crates.

Reply:
[frozen-abi-tree.txt](https://github.com/solana-labs/solana/files/11090220/frozen-abi-tree.txt) is the dependency tree of `frozen-abi` after removing unused ones.
Does not show any transitive dependency on `blake3` or `byteorder`.

Are these still needed after https://github.com/solana-labs/solana/pull/29893 which is using `{ workspace = true }` for dependencies version specification?

Reply:
Maybe they should be added as dependencies to another common dependency. You can remove them from frozen-abi crate.  I'm not sure how `{ workspace = true }` affects the original problem I was trying to solve. The crates in `programs/sbf/*` are not built as members of a workspace. When they are built for the sbf-solana-solana target they all define `global_allocator` by using the `entrypoint` macro https://github.com/solana-labs/solana/blob/29f776c676ed4676f64bff927da87cdab0672878/sdk/src/entrypoint.rs#L17  and cargo issues an error stating that only one `global_allocator` may be defined.

Please, feel free to remove the dependencies that you deem wrong.